engine/apps/slack/tasks.py

```diff
@@ -32,39 +32,102 @@
 
 
 @shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True)
-def update_incident_slack_message(slack_team_identity_pk, alert_group_pk):
-    cache_key = get_cache_key_update_incident_slack_message(alert_group_pk)
-    cached_task_id = cache.get(cache_key)
-    current_task_id = update_incident_slack_message.request.id
-
-    if cached_task_id is None:
-        update_task_id = update_incident_slack_message.apply_async(
-            (slack_team_identity_pk, alert_group_pk),
-            countdown=10,
+def update_alert_group_slack_message(slack_message_pk: int) -> None:
+    """
+    Background task to update the Slack message for an alert group.
+
+    This function is intended to be executed as a Celery task. It performs the following:
+    - Compares the current task ID with the `active_update_task_id` stored in the `SlackMessage`.
+      - If they do not match, it means a newer task has been scheduled, so the current task exits to prevent outdated updates.
+    - Does the actual update of the Slack message.
+    - Upon successful completion, clears the `active_update_task_id` to allow future updates.
+
+    Args:
+        slack_message_pk (int): The primary key of the `SlackMessage` instance to update.
+    """
+    from apps.slack.models import SlackMessage
+
+    current_task_id = update_alert_group_slack_message.request.id
+
+    logger.info(
+        f"update_alert_group_slack_message for slack message {slack_message_pk} started with task_id {current_task_id}"
+    )
+
+    try:
+        slack_message = SlackMessage.objects.get(pk=slack_message_pk)
+    except SlackMessage.DoesNotExist:
+        logger.warning(f"SlackMessage {slack_message_pk} doesn't exist")
+        return
+
+    if current_task_id != slack_message.active_update_task_id:
```

Comment:
I kinda liked how things were working with cache & timeout, especially:
```python3
cache.set(cache_key, update_task_id, timeout=CACHE_UPDATE_INCIDENT_SLACK_MESSAGE_LIFETIME)
```

this guarantees that even if things go wrong when setting/resetting the "lock", after X seconds it goes blank and Slack message will eventually get updated at some point.

but if we do:
```python3
def caller(...):
    task_id = celery_uuid()
    update_alert_group_slack_message.apply_async((self.pk,), countdown=countdown, task_id=task_id)
    self.active_update_task_id = task_id
    self.save(update_fields=["active_update_task_id"])

def task(...):
    if current_task_id != slack_message.active_update_task_id:
        return
    ...
    slack_message.active_update_task_id = None
    slack_message.save()
```

there's a non-zero chance of things going wrong wrt to concurrency (this chance will definitely materialize when working with a big enough number of Slack messages), for example in this scenario:
```python3
update_alert_group_slack_message.apply_async((self.pk,), countdown=countdown, task_id=task_id)
# imagine this task runs before `active_update_task_id` is updated in the DB, then the task will return early and not set `active_update_task_id` back to None

self.active_update_task_id = task_id
# after that, this change is persisted forever to the DB
# this means that any subsequent call to `update_alert_groups_message` will return early and this Slack message will never be updated anymore.
```

there's also a bunch of different ways things can go wrong, e.g. when multiple tasks run at the same time, celery worker crashing midway through running the task, retries, etc. this all can result in unpredictable behaviour and Slack message not ever updating.

having said all that, maybe we do something similar to https://github.com/grafana/oncall/blob/93c075098cef8d612a7f9dee3fe627267749c4ac/engine/common/utils.py#L91-L109 but without releasing the lock manually and just relying on redis timeout? this way we can guarantee the task runs only once in every X seconds (+ locks will definitely get cleared because redis does that automatically).

let me know if this all makes sense 😄